docs/guides/references/migration-guide.mdx

```diff
@@ -8,15 +8,131 @@ This guide details the changes and how to change your code to migrate to Cypress
+  const a = {}
+  setTimeout(() => {
+    a.foo = true
+  }, 1000)
+
+  return a
+}).should('have.property', 'foo')
+```
+
+The callback provided to `.then()` is only executed once. In Cypress 12, this
+command would fail immediately, because `.then()` only checked assertions once,
+but beginning in Cypress 13, `.should()` retries, waiting for the property to
+exist.
+
+#### It is safe to chain further queries after assertions
+
+```js
+cy.get('form').should('have.class', 'ready').find('button').click()
+```
+
+This test waits for a form to exist and have the "ready" class before clicking
+on a button inside of it. In Cypress 12, this test was vulnerable to detached
+DOM errors, where if React or another javascript library replaced elements on
+the page, Cypress could 'lose track' of them. In Cypress 13, this operation is
+reliable - because `.get().should().find()` is composed entirely of query
+commands, the whole chain will be retried while while trying to find and click
+on a button.
+
+#### Assertions are now included in aliases
+
+Previously, assertions were non-query commands, meaning that if `.as()` was used
+after them, they did not rerun, and the alias was vulnerable to the element
+becoming detached from the DOM if the page updated.
+
+```js
+cy.get('.modal').should('be.visible').as('myModal')
+
+cy.get('@myModal').invoke('hide')
+cy.get('@myModal').should('not.be.visible')
+```
+
+Beginning in Cypress 13, the above test will always fail. `.as()` is aliasing
+`.get('.modal').should('be.visible')`, which re-runs whenever the alias is read.
+This makes the last line impossible, since an element can never be both visible
+and not visible at the same time.
+
+You can resolve this by moving any assertions you don't want to re-run after
+`.as()`:
+
+```diff
+- cy.get('.modal').should('be.visible').as('myModal')
++ cy.get('.modal').as('myModal').should('be.visible')
+```
+
+#### Chains that include assertions now retry "all the way through"
+
+Previously, assertions were non-query commands, meaning that they served as
+"breakpoints" in a chain of queries, and queries before them did not re-run.
+
+```js
+cy.get('button')
+  .should('have.class', 'animating')
+  .get('button')
+  .should('have.class', 'done-animating')
+```
+
+In Cypress 12, this test would probably pass; because `.should()` was not a
+query, it broke the chain, and the above test could be thought of as having two
+parts:
+
+1. `.get(.button').should('have.class', 'animating')`
+2. `.get('button').should('have.class', 'done-animating')`
+
+However, in Cypress 13, the chain is composed entirely of queries, meaning the
+entire chain of all four commands must pass at the same time.
+
+You can resolve this by breaking up chains of commands that you don't want to
+execute together with `cy`.
+
+```diff
+-   .get('button')
++ cy.get('button')
+```
+
+See the [Retry-Ability Guide](/guides/core-concepts/retry-ability) for more
+details on how queries chain together and retry.
+
```

Comment:
Should there also be a section about not being able to overwrite `.should()` and `.and()` with `Cypress.Commands.overwrite()` anymore, similar to the one for `.readFile()`?

Reply:
changes should be removed due to https://github.com/cypress-io/cypress/pull/25738